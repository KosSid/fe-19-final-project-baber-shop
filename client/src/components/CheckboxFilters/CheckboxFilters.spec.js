import React from 'react';
import CheckboxFilter from "./index";
import {render, unmountComponentAtNode} from 'react-dom'
import {act} from '@testing-library/react'
import {MemoryRouter} from "react-router";
import renderer from "react-test-renderer";

let container = null;
beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
})

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
})

test('CheckboxFilter is rendered', () => {
    act(() => {
        render(<MemoryRouter>
            <CheckboxFilter/>
        </MemoryRouter>, container)
    });

    expect(container.innerHTML).toMatchSnapshot();
});

test('renders correctly when there is one filter', () => {
    const items = ['filter'];
    const tree = renderer.create(<CheckboxFilter items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly when there are couple of filters', () => {
    const items = ['filter-one', 'filter-two', 'filter-three'];
    const tree = renderer.create(<CheckboxFilter items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
});