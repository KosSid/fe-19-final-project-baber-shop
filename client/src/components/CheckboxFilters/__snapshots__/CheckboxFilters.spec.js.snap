// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxFilter is rendered 1`] = `"<div class=\\"checkbox-container\\"><div class=\\"checkbox-group checkbox-catalog-group\\"><p class=\\"checkbox-group__name\\">Catalog</p></div></div>"`;

exports[`renders correctly when there are couple of filters 1`] = `
<div
  className="checkbox-container"
>
  <div
    className="checkbox-group checkbox-catalog-group"
  >
    <p
      className="checkbox-group__name"
    >
      Catalog
    </p>
  </div>
</div>
`;

exports[`renders correctly when there is one filter 1`] = `
<div
  className="checkbox-container"
>
  <div
    className="checkbox-group checkbox-catalog-group"
  >
    <p
      className="checkbox-group__name"
    >
      Catalog
    </p>
  </div>
</div>
`;
